core/metrics-core-service/src/main/java/org/hawkular/metrics/core/dropwizard/HawkularMetricsRegistryListener.java

```diff
@@ -68,56 +70,147 @@ public void setMetricNameService(MetricNameService metricNameService) {
-                new Metric<>(new MetricId<>(tenantId, COUNTER, metricName + "-95p"), metaData.getTags()),
-                new Metric<>(new MetricId<>(tenantId, COUNTER, metricName + "-99p"), metaData.getTags()),
-                new Metric<>(new MetricId<>(tenantId, COUNTER, metricName + "-999p"), metaData.getTags())
+                new Metric<>(new MetricId<>(tenantId, COUNTER, metricName), metaData.getTags()),
+                new Metric<>(new MetricId<>(tenantId, GAUGE, metricName + "-median"), metaData.getTags()),
+                new Metric<>(new MetricId<>(tenantId, GAUGE, metricName + "-max"), metaData.getTags()),
+                new Metric<>(new MetricId<>(tenantId, GAUGE, metricName + "-min"), metaData.getTags()),
+                new Metric<>(new MetricId<>(tenantId, GAUGE, metricName + "-75p"), metaData.getTags()),
+                new Metric<>(new MetricId<>(tenantId, GAUGE, metricName + "-95p"), metaData.getTags()),
+                new Metric<>(new MetricId<>(tenantId, GAUGE, metricName + "-99p"), metaData.getTags()),
+                new Metric<>(new MetricId<>(tenantId, GAUGE, metricName + "-999p"), metaData.getTags())
+
+        ));
+    }
+
+    @Override
+    public void onTimerRemoved(String name) {
+        MetaData metaData = metricRegistry.removeMetaData(name);
+        if (metaData == null) {
+            logger.warnf("Did not find meta data for timer %s that has been removed from the registry");
+            return;
+        }
+
+        String tenantId = metricNameService.getTenantId();
+        String metricName = metricNameService.createMetricName(metaData);
+
+        deleteMetrics(metricName, asList(
+                new MetricId<>(tenantId, GAUGE, metricName + "-5min"),
+                new MetricId<>(tenantId, GAUGE, metricName + "-15min"),
+                new MetricId<>(tenantId, GAUGE, metricName + "-mean"),
+                new MetricId<>(tenantId, COUNTER, metricName),
+                new MetricId<>(tenantId, GAUGE, metricName + "-median"),
+                new MetricId<>(tenantId, GAUGE, metricName + "-max"),
+                new MetricId<>(tenantId, GAUGE, metricName + "-min"),
+                new MetricId<>(tenantId, GAUGE, metricName + "-75p"),
+                new MetricId<>(tenantId, GAUGE, metricName + "-95p"),
+                new MetricId<>(tenantId, GAUGE, metricName + "-99p"),
+                new MetricId<>(tenantId, GAUGE, metricName + "-999p")
+        ));
+    }
+
+    @Override
+    public void onHistogramAdded(String name, Histogram histogram) {
+        MetaData metaData = metricRegistry.getMetaData(name);
+        if (metaData == null) {
+            logger.warnf("Did not find meta data for histogram %s that has been added to the registry", name);
+            return;
+        }
+
+        String tenantId = metricNameService.getTenantId();
+        String metricName = metricNameService.createMetricName(metaData);
+
+        createMetrics(metricName, asList(
```

Comment:
To be honest, I am not entirely sure what we want to store from the aggregated DropWizard metrics like timers and meters. I do not however think we need to figure that out in this PR. We could just store the raw gauge and counter values, but then that begs the question of why use the DropWizards metrics library at all.

 I think there is value in storing what are essentially pre-computed percentiles. We can easily view graphs of those percentiles with the Grafana plugin or the new UI being developed. We can also compute the min, max, avg of p99 for instance. We can make changes to do these things ad hoc, but it is not currently possible.